### ISSUE TITLE
Geofence_service breaks local notifications on iOS

### ISSUE THREAD

**Issue by NedasZilovas** [Dev-hwang/geofence_service#12]
When using geofence_service together with flutter_local_notifications, when you minimize the app on an iOS device, the local notifications no longer trigger methods on notification clicks.

The breaking point on this issue (as far as I can tell) is on the app minimize, notification clicks trigger correctly if you start the app and never minimize it or if you stop the geofence_service before the first app minimize. 

After the app minimizes once with the geofence_service on, I can no longer get the notifications to trigger any methods on click.

Steps to reproduce:
1) get the example app from flutter_local_notifications
2) add geofence_service package
3) set it up so the geofence_service and the foreground task starts on the first screen of the app
3.1*) there appears an error on one of the notification showcases when you add geofence_service package, can comment over that method since it wont be used for the test.
4) start up the app on an iOS device
5) send a notification via the app and click it. This should redirect to 2nd page of the app (works correctly)
6) minimize the app and go back to it
7) send a notification via the app and click it -> nothing happens (notification breaks)

Any help on this would be greatly appreciated.

**Comment by Dev-hwang** [comment]
There seems to be a feature conflict when using the singleton object `UNUserNotificationCenter.current()` concurrently on the iOS platform. This can be temporarily resolved as follows. I'll let you know when the problem is resolved.

```dart
  @override
  Widget build(BuildContext context) {
    Widget child = Scaffold(
      appBar: AppBar(
        title: const Text('Plugin example app'),
      ),
      body: Container(),
    );

    if (Platform.isAndroid) {
      child = WillStartForegroundTask(
        onWillStart: () async {
          // You can add a foreground task start condition.
          return _geofenceService.isRunningService;
        },
        androidNotificationOptions: const AndroidNotificationOptions(
          channelId: 'geofence_service_notification_channel',
          channelName: 'Geofence Service Notification',
          channelDescription: 'This notification appears when the geofence service is running in the background.',
          channelImportance: NotificationChannelImportance.LOW,
          priority: NotificationPriority.LOW,
          isSticky: false,
        ),
        // iosNotificationOptions: const IOSNotificationOptions(),
        notificationTitle: 'Geofence Service is running',
        notificationText: 'Tap to return to the app',
        child: child,
      );
    }

    return MaterialApp(home: child);
  }
```

**Comment by NedasZilovas** [comment]
Unfortunately the temporary solution isn't quite fitting for the project my team is working on as the geofence service is more or less an essential part of the app, along with the notifications.

So I would like to ask, is this an issue you would be looking to resolve in the short term? (and if so, would it be possible to know the estimated time? )

**Comment by Dev-hwang** [comment]
It doesn't seem like a problem that can be solved in a short period of time. The `geofence service` and the `WillStartForegroundTask` widget with notification functions are separate. Therefore, it is recommended to implement the `WillStartForegroundTask` widget with notification function separately or use another plugin.

**Comment by Dev-hwang** [comment]
In version `3.4.6`, this issues has been fixed.

1. Upgrade dependencies.

```yaml
dependencies:
  geofence_service: ^3.4.6
```

2. To display a notification when your app enters the background, you need to open the `ios/Runner/AppDelegate` file and set the following:

**Objective-C**:

```objectivec
@implementation AppDelegate

- (BOOL)application:(UIApplication *)application
    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
  [GeneratedPluginRegistrant registerWithRegistry:self];

  // here
  if (@available(iOS 10.0, *)) {
    [UNUserNotificationCenter currentNotificationCenter].delegate = (id<UNUserNotificationCenterDelegate>) self;
  }

  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}

@end
```

**Swift**:

```swift
@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    GeneratedPluginRegistrant.register(with: self)

    // here
    if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }

    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}
```

**Comment by Dev-hwang** [comment]
If the problem persists despite upgrading the plugin, please create a new issue.
